src/main/java/cloud/appuio/keycloak/extensions/mappers/GroupNameFormatter.java

```diff
@@ -0,0 +1,67 @@
+package cloud.appuio.keycloak.extensions.mappers;
+
+import org.keycloak.provider.ProviderConfigProperty;
+
+public class GroupNameFormatter {
+
+    private boolean trimWhitespace;
+    private boolean toLowerCase;
+    private String trimPrefix = "";
+
+    GroupNameFormatter() {
+    }
+
+    String format(String input) {
+        if (input == null) return "";
+        var s = input;
+        if (!trimPrefix.equals("")) {
+            s = s.replaceFirst(trimPrefix, "");
+        }
+        if (trimWhitespace) {
+            s = s.trim().replaceAll("\\s+", "-").replaceAll("(--)+", "-");
```

Comment:
Actually, I think `replaceAll("-+", "-")` should be sufficient (not sure if it's more efficient than `--+` though), looking at it again, since you always want to replace any sequence of dashes with a single dash.